lib/browser/api/dialog.js

```diff
@@ -70,6 +70,35 @@ const checkAppInitialized = function () {
   }
 }
 
+const saveDialog = (sync, window, options) => {
+  checkAppInitialized()
+
+  if (window.constructor !== BrowserWindow) options = window
+  if (options == null) options = { title: 'Save' }
+
+  const {
+    buttonLabel = '',
+    defaultPath = '',
+    filters = [],
+    title = '',
+    message = '',
+    securityScopedBookmarks = false,
+    nameFieldLabel = '',
+    showsTagField = true
+  } = options
+
+  if (typeof title !== 'string') throw new TypeError('Title must be a string')
+  if (typeof buttonLabel !== 'string') throw new TypeError('Button label must be a string')
+  if (typeof defaultPath !== 'string') throw new TypeError('Default path must be a string')
+  if (typeof message !== 'string') throw new TypeError('Message must be a string')
+  if (typeof nameFieldLabel !== 'string') throw new TypeError('Name field label must be a string')
+
+  const settings = { buttonLabel, defaultPath, filters, title, message, securityScopedBookmarks, nameFieldLabel, showsTagField, window }
+
+  if (sync) return binding.showSaveDialogSync(settings)
+  binding.showSaveDialog(settings)
```

Comment:
missing return

Reply:
IMO this would be more readable
```
  if (sync) {
    return binding.showSaveDialogSync(settings)
  } else {
    return binding.showSaveDialog(settings)
  }
```